feat(curriculum): adding shape manager TS workshop (#63083)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-shape-manager/6915a808bf21675d5640a4e3.md

---
id: 6915a808bf21675d5640a4e3
title: Step 33
challengeType: 0
dashedName: step-33
---

# --description--

> Now, you should select the shapes. Each of them are in a `div` with the id `[shape]-props`. Instead of selecting them one after the other, reassign the `propertyGroups` inline type with `circle`, `rectangle`, and `triangle` properties.

Reviewer: ```suggestion
Now, you should select the shapes. Each of them is in a `div` with the id `[shape]-props`. Instead of selecting them one after the other, reassign the `propertyGroups` inline type with `circle`, `rectangle`, and `triangle` properties.
```

Each of the properties should use the `getElements` function with `HTMLElement` type to get elements with ids `circle-props`, `rectangle-props`, and `triangle-props` respectively.

# --hints--

> You should reassign the `propertyGroups` inline type variable with the appropriate values.

Reviewer: ```suggestion
You should assign an object to the `propertyGroups` variable declared before.
```
